docker/bootstrap

```diff
@@ -0,0 +1,47 @@
+#!/bin/sh
+
+set -e
+
+cd "$(dirname "$0")/.."
+
+error_msg() { printf "\033[31m%s\033[0m\n" "$*"; }
+notice_msg() { printf "\033[33m%s\033[0m " "$*"; }
+success_msg() { printf "\033[32m%s\033[0m\n" "$*"; }
+
+notice_msg "Checking dependencies..."
+if ! which docker &>/dev/null; then
+  error_msg 'Docker command not found. Please install: https://docs.docker.com/get-docker/'
+  exit 1
+fi
+
+required_docker="20"
+current_docker="$(docker --version | awk 'match($0, /[0-9\.]+/) {print substr($0,RSTART,RLENGTH)}')"
+if [ "$(printf '%s\n' "$required_docker" "$current_docker" | sort -V | head -n1)" != "$required_docker" ]; then
+  error_msg "Requires Docker v$required_docker or later. Please upgrade and rerun."
+  exit 1
+fi
+
+themes_dir="$(dirname $(pwd))/alaveteli-themes"
+if [ ! -d $themes_dir ]; then
+  error_msg "Can't find $themes_dir directory. Please create and rerun."
+  exit 1
+fi
+success_msg 'done'
+
+notice_msg "Copying example files..."
+[ ! -f config/database.yml ] && cp config/database.yml-example config/database.yml
+[ ! -f config/general-alavetelitheme.yml ] && cp config/general.yml-example config/general-alavetelitheme.yml
+[ ! -L config/general.yml ] && [ ! -f config/general.yml ] && ln -s config/general-alavetelitheme.yml config/general.yml
+success_msg 'done'
+
+notice_msg "Syncing git submodules..."
+git submodule update --init
+success_msg 'done'
+
+notice_msg "Getting latest alavetelitheme commits..."
+alavetelitheme="$themes_dir/alavetelitheme"
+[ ! -d $alavetelitheme ] && git clone https://github.com/mysociety/alavetelitheme.git $alavetelitheme >/dev/null
+pushd $alavetelitheme >/dev/null && git pull >/dev/null && popd >/dev/null
+success_msg 'done'
+
+./docker/setup
```

Comment:
Just a suggestion, but I think it would be closer to the scripts-to-rule-them-all pattern to be able to run `docker/bootstrap` without also calling `docker/setup`. Personally speaking I expect the bootstrap script simply to deal with pre-requisites for running things, and for that script to then be called elsewhere where necessary.

I suggest replacing this line will a call to `docker/bootstrap` inside the `if [ -z "$DOCKER ]; then` block in `docker/setup`.